feat: Add atomic values in API messages

regseb opened the pull request:
Fix https://github.com/birjj/svglint/issues/91

ericcornelissen reviewed (commented):
Changes make sense to me. Any chance there's a test that can be updated (or added) to verify the changes? Else, could you provide me with some steps to verify the changes manually?

regseb commented:
I've added tests for the `Reporter` class. I've also updated the `expect` dependency to have [`.toMatchObject(object)`](https://jestjs.io/docs/expect#tomatchobjectobject) method.

ericcornelissen commented on package.json:
```diff
@@ -52,7 +52,7 @@
     "eslint": "^8.3.0",
     "execa": "^6.0.0",
-    "expect": "^1.20.2",
+    "expect": "^29.7.0",
```
Can you change this to `"^28.1.3"` so that it can run on Node.js v12, that should fix [this error](https://github.com/birjj/svglint/actions/runs/6676161047/job/18161511215).

regseb replied:
There's another [problem](https://github.com/birjj/svglint/actions/runs/6692359102/job/18181695402?pr=92#step:5:9) with Node.js `v16.0.0`. I can change the version in workflow [`test.yml`](https://github.com/birjj/svglint/blob/33b56c2/.github/workflows/test.yml#L33) to `16.10.0`?

```
Run npm ci
npm ERR! code EBADENGINE
npm ERR! engine Unsupported engine
npm ERR! engine Not compatible with your version of node/npm: @jest/expect-utils@28.1.3
npm ERR! notsup Not compatible with your version of node/npm: @jest/expect-utils@28.1.3
npm ERR! notsup Required: {"node":"^12.13.0 || ^14.15.0 || ^16.10.0 || >=17.0.0"}
npm ERR! notsup Actual:   {"npm":"8.1.2","node":"v16.0.0"}

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2023-10-30T12_03_58_308Z-debug.log
Error: Process completed with exit code 1.
```

ericcornelissen replied:
Since this project aims to be [compatible with 16.0.0](https://github.com/birjj/svglint/blob/834abc486612349a9d4f30e8d7990506ddc0d96c/package.json#L60) I'd prefer to keep the CI test against that version. I checked once more and I believe `"^27.5.1"` should resolve the issue for Node.js v16.0.0 as well. From the [npm code explorer](https://www.npmjs.com/package/expect/v/27.5.1?activeTab=code) that version should work with `"node": "^10.13.0 || ^12.13.0 || ^14.15.0 || >=15.0.0"`.

ericcornelissen commented on test/reporter.spec.js:
```diff
@@ -0,0 +1,66 @@
+            { elm: { "g > path": false } }
+        );
+        expect(results.elm.messages).toMatchObject([{
+            message: "Element disallowed\n  At node \u{1B}[1m<path>\u{1B}[22m (2:17)",
```
I think we can just use [`chalk`](https://www.npmjs.com/package/chalk) here, like in [the source](https://github.com/birjj/svglint/blob/834abc486612349a9d4f30e8d7990506ddc0d96c/src/lib/reporter.js#L44), to ensure ANSI styling is applied consistently between the implementation and test assertions. That should fix the remaining [CI errors](https://github.com/birjj/svglint/actions/runs/6676161047/job/18161511290).

ericcornelissen commented:
Thank you very much for your contribution @regseb!
